File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
import java.util.*; 
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        List<Integer> dis1 = new ArrayList<Integer>();
        List<Integer> dis2 = new ArrayList<Integer>();     
        HashSet<Integer> hs1 = new HashSet<Integer>();   
        HashSet<Integer> hs2 = new HashSet<Integer>();   
        for (int n : nums1){
             if (!hs1.contains(n)){
                hs1.add(n);
             }
        }
        for (int n: nums2){
             if (!hs2.contains(n)){
                hs2.add(n);
             }
        }

        for (int n : hs1){
            if (!hs2.contains(n)){
                dis1.add(n);
            }
        }
        for (int n : hs2){
            if (!hs1.contains(n)){
                dis2.add(n);
            }
        }
        List<List<Integer>> output= new ArrayList<List<Integer>>();
        output.add(dis1);
        output.add(dis2); 
        return output;
    }
}

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cs
using System;
using System.Collections.Generic;

public class Solution {
    public IList<IList<int>> FindDifference(int[] nums1, int[] nums2) {
        List<int> dis1 = new List<int>();
        List<int> dis2 = new List<int>();     
        HashSet<int> hs1 = new HashSet<int>();   
        HashSet<int> hs2 = new HashSet<int>();   
        
        foreach (int n in nums1) {
            if (!hs1.Contains(n)) {
                hs1.Add(n);
            }
        }
        
        foreach (int n in nums2) {
            if (!hs2.Contains(n)) {
                hs2.Add(n);
            }
        }

        foreach (int n in hs1) {
            if (!hs2.Contains(n)) {
                dis1.Add(n);
            }
        }

        foreach (int n in hs2) {
            if (!hs1.Contains(n)) {
                dis2.Add(n);
            }
        }
        
        IList<IList<int>> output = new List<IList<int>>();
        output.Add(dis1);
        output.Add(dis2); 
        return output;
    }
}
